marxs/optics/multiLayerMirror.py

```diff
@@ -92,14 +93,17 @@ def process_photons(self, photons):
         spread_refl = np.interp(local_intersection[:,1], local_coords_in_file, reflectFile['FWHM(nm)'])
         
         wavelength = 1.23984282 / photons['energy']   # wavelength is in nm assuming energy is in keV
-        c_squared = (spread_refl ** 2) / (2 * np.log(2))
-        refl_prob = max_refl * np.exp(-((wavelength - peak_wavelength) ** 2) / (2 * c_squared))
+        c_squared = (spread_refl ** 2) / (8. * np.log(2))
+        c_is_zero = (c_squared == 0)
+    	
+        refl_prob = np.zeros(len(wavelength))
+        refl_prob[~c_is_zero] = max_refl[~c_is_zero] * np.exp(-((wavelength[~c_is_zero] - peak_wavelength[~c_is_zero]) ** 2) / (2 * c_squared[~c_is_zero]))
         
         # find probability of being reflected due to polarization
         # v_1 is parallel to plane, good reflection direction
-        refl_prob *= np.einsum('ij,ij->i', polarization, v_1)**2
+        refl_prob[~c_is_zero] *= np.einsum('ij,ij->i', polarization[~c_is_zero], v_1[~c_is_zero]) ** 2
         refl_prob[np.isnan(refl_prob)] = 0
         # multiply probability by probability of reflection
         photons['probability'] *= refl_prob / 100
         
-        return photons
+        return photons[photons['probability'] != 0]
```

Comment:
I added that to reduce the amount of time it would take because it's starting to get slow. With a few different optics in a script, I think this will make a big difference. We can add in a calculation of the number of photons at the beginning if that's the only thing that would be lost. Is it ok if I leave that in for now? It's possible that later other speed improvements could make this unnecessary. One alternative is that for the mirror at the top I can add something like:
`photons_remaining = all_photons['probability'] > 0`
`photons = all_photons[photons_remaining]`
and then at the end:
`all_photons[photons_remaining] = photons`
`return all_photons`